### PR TITLE
PS-5326 8.0.14 should be able to decrypt binary logs encrypted with

### PR DESCRIPTION
**Pull request by robgolebiowski**
8.0.13

5.7 binlog encryption was removed. The part resposible for decrypting
5.7 encrypted binary log was kept. binlog_encryption test suite was
renamed to bin_57_decryption. The tests in this test suite were
rewritten to test 5.7 decryption, some of the tests were removed.
Also almost all tests from rpl_encryption test suite were removed,
apart from the tests that are "greping" the files that could contain
sensitive information to check whether they are encrypted.
The START_ENCRYPTION_EVENT was changed to START_5_7_ENCRYPTION_EVENT
and error messages were amended accordingly.

### CODE REVIEW

**Comment by robgolebiowski** [comment]
https://ps2.cd.percona.com/view/8.0/job/percona-server-8.0-param/52/

File: mysql-test/mysql-test-run.pl [hunk 1]
```diff
@@ -159,7 +159,7 @@
 my $path_vardir_trace;      # Unix formatted opt_vardir for trace files
 
 my $DEFAULT_SUITES =
-"main,sys_vars,binlog,binlog_gtid,binlog_nogtid,federated,gis,rpl,rpl_gtid,rpl_nogtid,innodb,innodb_gis,innodb_fts,innodb_zip,innodb_undo,perfschema,funcs_1,opt_trace,parts,auth_sec,query_rewrite_plugins,gcol,sysschema,test_service_sql_api,json,connection_control,test_services,collations,service_udf_registration,service_sys_var_registration,service_status_var_registration,x,secondary_engine,"
+"main,sys_vars,binlog,binlog_gtid,binlog_nogtid,bin_57_decryption,rpl_encryption,federated,gis,rpl,rpl_gtid,rpl_nogtid,innodb,innodb_gis,innodb_fts,innodb_zip,innodb_undo,perfschema,funcs_1,opt_trace,parts,auth_sec,query_rewrite_plugins,gcol,sysschema,test_service_sql_api,json,connection_control,test_services,collations,service_udf_registration,service_sys_var_registration,service_status_var_registration,x,secondary_engine,"
```
Comment (laurynas-biveinis) [inline mysql-test/mysql-test-run.pl]: It's good to have full "binlog" substring in the suite name:
```suggestion
"main,sys_vars,binlog,binlog_gtid,binlog_nogtid,binlog_57_decryption,rpl_encryption,federated,gis,rpl,rpl_gtid,rpl_nogtid,innodb,innodb_gis,innodb_fts,innodb_zip,innodb_undo,perfschema,funcs_1,opt_trace,parts,auth_sec,query_rewrite_plugins,gcol,sysschema,test_service_sql_api,json,connection_control,test_services,collations,service_udf_registration,service_sys_var_registration,service_status_var_registration,x,secondary_engine,"
```

File: mysql-test/suite/encryption/t/percona_system_key_rotation.test [hunk 2]
```diff
@@ -41,10 +37,14 @@
 --let $assert_cond= "[SELECT rotate_system_key\\(\\"123\\"\\) = 0]" = 1
 --source include/assert.inc
 
+--let $assert_text= It should BE possible to rotate key percona_innodb
```
Comment (laurynas-biveinis) [inline mysql-test/suite/encryption/t/percona_system_key_rotation.test]:
```suggestion
--let $assert_text= It should be possible to rotate key percona_innodb
```

File: mysql-test/suite/rpl/t/percona_rpl_encryption_master_binlog_ps_encrypted.test [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+#    INSERT INTO t1 (a) VALUES (1);
+#    INSERT INTO t1 (a) VALUES (2);
+#    INSERT INTO t1 (a) VALUES (3);
+#    INSERT INTO t1 (a) VALUES (4);
+#    INSERT INTO t1 (a) VALUES (5);
+# 2) prepared ps_encrypted_truncated_master-bin.000002. It has the same statements as 1), however the last event is incomplete -
+#    and was manually truncated; (encryption key is in keyring_with_percona_binlog_key)
+
+# In this test we will test following scenarios:
+# 1) Generate two master-bin on server (without encrypteion), replace master-bin.000002 with ps_encrypted_truncated_master_bin.000002.
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rpl/t/percona_rpl_encryption_master_binlog_ps_encrypted.test]:
```suggestion
# 1) Generate two master-bin on server (without encryption), replace master-bin.000002 with ps_encrypted_truncated_master_bin.000002.
```

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
No need for further reviews

**Pull request merged**